lms/djangoapps/grades/new/course_grade.py

```diff
@@ -123,30 +152,24 @@ def compute_and_update(self, read_only=False):
 
         If read_only is True, doesn't save any updates to the grades.
         """
-        subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
-        subsections_total = 0
-        for chapter_key in self.course_structure.get_children(self.course.location):
-            chapter = self.course_structure[chapter_key]
-            chapter_subsection_grades = []
-            children = self.course_structure.get_children(chapter_key)
-            subsections_total += len(children)
-            for subsection_key in children:
-                chapter_subsection_grades.append(
-                    subsection_grade_factory.create(self.course_structure[subsection_key], read_only=True)
-                )
-
-            self.chapter_grades.append({
-                'display_name': block_metadata_utils.display_name_with_default_escaped(chapter),
-                'url_name': block_metadata_utils.url_name_for_block(chapter),
-                'sections': chapter_subsection_grades
-            })
+        subsections_total = sum([len(chapter['sections']) for chapter in self.chapter_grades])
```

Comment:
Like the sum that follows this, you don't need to actually create a list before you sum it.  Save memory by dropping the square brackets, so you get a generator instead of a list.